### ISSUE TITLE
Get rid of references to maintainer email addresses from code

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2438]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#2438|
|Type|enhancement|
|Created|2007-02-06 14:25:10Z|
|Branch|https://github.com/twisted/twisted/tree/maintainer-email-2438|

References to my twisted (at) itamarst.org email should be removed, just leaving my name (I'm easy to google.) I get so much spam on that email account that I ended up just sending everything to spam folder, and anyway people should be adding stuff to issue tracker, not emailing me directly.

Attachments:

* [itamarst-email-2438.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/457/45768a7f894545ca94235b5eede7baa0841db493/34b4bfc942027c17fde29f84995ddf9532e74ae9.patch) (12391 bytes) - added by collab on 2008-07-01 19:41:48Z - patch against r24199
* [maintainer-email-2438.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/457/45768a7f894545ca94235b5eede7baa0841db493/0edef5668c4adba379002bfc3e1000109a70d3f3.patch) (55716 bytes) - added by collab on 2008-07-03 22:05:56Z - patch against r24235

<details><summary>Searchable metadata</summary>

```
trac-id__2438 2438
type__enhancement enhancement
reporter__itamarst itamarst
priority__low low
milestone__ 
branch__branches_maintainer_email_2438 branches/maintainer-email-2438
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1170771910000000 1170771910000000
changetime__1217362451000000 1217362451000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|collab set owner to collab<br>collab set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|collab removed owner<br>collab set status to `new`|
|-|-|

Attached a [[itamarst-email-2438.patch](None/itamarst-email-2438.patch) patch] that fixes the issue.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to collab|
|-|-|

Thanks for the patch.

The reasons that apply to Itamar probably apply to everyone else.  Can you work up a patch that drops the email from all the maintainer lines?

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|collab set status to `assigned`|
|-|-|

I'll add the others as well. Changing the title of the ticket  to reflect that update.

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|collab removed owner<br>collab set status to `new`|
|-|-|

Added a new [[maintainer-email-2438.patch](None/maintainer-email-2438.patch) patch] that includes everyone, hope I didn't miss anyone ;)

I didn't update the ones in the CREDITS or copyright files by the way, I wasn't sure about that (ticket title mentions code only) so I'll leave that decision up to you.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to collab|
|-|-|

`twisted/protocols/ftp.py` still has a couple addresses

`twisted/names/dns.py` could probably have the lines of its `@author` field folded

`twisted/trial/test/test_runner.py` still has an email address.

`twisted/internet/defer.py` should probably just remove the maintainer lines from `inlineCallbacks` and `deferredGenerator`

`twisted/protocols/htb.py`, `twisted/protocols/dict.py`, `twisted/conch/ui/tkvt100.py`, `twisted/conch/ui/ansi.py`, `twisted/words/protocols/irc.py`, `twisted/words/protocols/msn.py`, and `twisted/internet/epollreactor.py` also still have email addresses.  So do a lot of files in `twisted/flow/`, but I don't think we need to update those (flow will be removed soon enough).

Thanks!

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|collab set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|collab removed owner<br>collab set status to `new`|
|-|-|

Replying to [exarkun](#note_6):
> `twisted/protocols/ftp.py` still has a couple addresses
>
Strange that I didn't catch those ones with the rpl tool I'm using.

> `twisted/names/dns.py` could probably have the lines of its `@author` field folded
>

Fixed.

> `twisted/trial/test/test_runner.py` still has an email address.
>
> `twisted/internet/defer.py` should probably just remove the maintainer lines from `inlineCallbacks` and `deferredGenerator`

Removed them.

>
> `twisted/protocols/htb.py`, `twisted/protocols/dict.py`, `twisted/conch/ui/tkvt100.py`, `twisted/conch/ui/ansi.py`, `twisted/words/protocols/irc.py`, `twisted/words/protocols/msn.py`, and `twisted/internet/epollreactor.py` also still have email addresses.  So do a lot of files in `twisted/flow/`, but I don't think we need to update those (flow will be removed soon enough).

I also ran a search for the author listed in flow and fixed that one as well. Added an updated [[maintainer-email-2438.patch](None/maintainer-email-2438.patch) patch].

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Looks good.  It'd be great to put this into a branch and run the documentation builder against it though, just to be sure everything's right.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [exarkun](#note_10):
> Looks good.  It'd be great to put this into a branch and run the documentation builder against it though, just to be sure everything's right.

Is the process of 'running the documentation builder against it' documented somewhere? Or is it as simple as running pydoctor?

Putting the new [branches/maintainer-email-2438 branch](branches/maintainer-email-2438 branch) up for review so someone can point me to the right docs and/or give me a hint.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

If you visit http://buildbot.twistedmatrix.com/boxes-supported and click on the "documentation" link near the top-left of the page, you'll be presented with a form for forcing a build on the documentation builder.  Enter the branch details and submit and then watch to see what the results are.

Also, this branch is wacky.  Combinator docs - http://divmod.org/trac/wiki/DivmodCombinator

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra commented|
|-|-|

(In [24426]) Branching to 'maintainer-email-2438'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [exarkun](#note_12):
> If you visit http://buildbot.twistedmatrix.com/boxes-supported and click on the "documentation" link near the top-left of the page, you'll be presented with a form for forcing a build on the documentation builder.  Enter the branch details and submit and then watch to see what the results are.

Looks good: http://buildbot.twistedmatrix.com/builders/documentation/builds/708

Putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_15"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_16"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24441]) Merge maintainer-email-2438: Get rid of references to maintainer email addresses from code.

Author: thijs
Reviewer: exarkun
Fixes: [#2438](https://github.com/twisted/twisted/issues/2438)
